### ISSUE TITLE
Error when attempting to signin with github

### ISSUE THREAD

**Issue by justinelsberry** [scala-exercises/scala-exercises#451]
After clicking "Login With Github" I am immediately returned to the site with the falling error and callback url: 

https://www.scala-exercises.org/_oauth-callback?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Foauth%2F%23redirect-uri-mismatch&state=126dc86a-6244-4106-9fdf-491a9327b791

![screen shot 2016-06-22 at 7 19 24 am](https://cloud.githubusercontent.com/assets/205617/16270098/ea75554a-3849-11e6-88bc-713ac5466d24.png)

**Comment by dialelo** [comment]
Should be related to the redirection from the top-level domain to the "www" subdomain, looking into it.

**Comment by rafaparadela** [comment]
@dialelo @justinelsberry I think I know why this issue has happened, because it have got the same message few minutes ago.

I wanted to answer a comment that has been added in this URL:

```
https://scala-exercises.herokuapp.com/std_lib/asserts#comment-2744423915
```
This is the URL of the Heroku app. So if you try to authenticate from there, GitHub oAuth will return you back to the callback-URL `https://www.scala-exercises.org/_oauth-callback`. Given that those redirect-urls don't match, this error message is returned (as querystring):

```
The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application
```

In summary: We should automatically redirect from `https://scala-exercises.herokuapp.com` to `https://www.scala-exercises.org`, and the issue will be fixed.

**Comment by rafaparadela** [comment]
In short: Authentication in `https://scala-exercises.herokuapp.com` is not enabled.

**Comment by rafaparadela** [comment]
In fact, the actual issue is related to the subdomain `www` as @dialelo has mentioned.

**Comment by rafaparadela** [comment]
Two possible solutions:
- Always redirect from `https://scala-exercises.org` to `https://www.scala-exercises.org`.
- Generate and send a valid `redirect_url` to Github agnostic to the subdomain.

**Comment by raulraja** [comment]
Fixed in #456
